$NetBSD: patch-ae,v 1.3 2007/09/28 21:11:50 dmcmahill Exp $

- add missing checks for -lnsl and -lsocket needed by at least
  SunOS-5.9 and possibly others.

- Fix the SUBDIRS stuff to work correctly.

- add missing AM_PROG_CC_C_O that automake wants for the yacc
  sources.

- allow a different default serial/parallel port to be specified
  which is useful for pkgsrc (since we have defaults for various
  platforms available and they're not the same for different
  MACHINE_ARCH's in NetBSD).

--- configure.ac.orig	2006-10-09 10:47:29.000000000 -0400
+++ configure.ac	2007-09-28 12:55:21.429495000 -0400
@@ -45,4 +45,6 @@
 AC_CHECK_LIB([ncurses], [tputs])
 AC_CHECK_LIB([readline], [readline])
+AC_CHECK_LIB([nsl], [gethostbyname])
+AC_CHECK_LIB([socket], [socket])
 AH_TEMPLATE([HAVE_LIBUSB],
             [Define if USB support is enabled via libusb])
@@ -67,4 +69,5 @@
 # Checks for typedefs, structures, and compiler characteristics.
 AC_C_CONST
+AM_PROG_CC_C_O
 AC_HEADER_TIME
 
@@ -119,48 +122,48 @@
 
 if test "$enabled_doc" = "yes"; then
-
-SUBDIRS_AC='doc @WINDOWS_DIRS@'
-DIST_SUBDIRS_AC='doc windows'
+	DOC_DIR="doc"
 else
-
-SUBDIRS_AC='@WINDOWS_DIRS@'
-DIST_SUBDIRS_AC='windows'
+	DOC_DIR=""
 fi
 
 AC_SUBST(DOC_INST_DIR, $DOC_INST_DIR)
-AC_SUBST(SUBDIRS_AC, $SUBDIRS_AC)
-AC_SUBST(DIST_SUBDIRS_AC, $DIST_SUBDIRS_AC)
-
+AC_SUBST(DOC_DIR, $DOC_DIR)
 
 # Find the parallel serial device files based on target system
 # If a system doesn't have a PC style parallel, mark it as unknown.
+#
+# Allow the user to set the value of both
 case $target in
 	i[[3456]]86-*-linux*|x86_64-*-linux*)
-		DEFAULT_PAR_PORT="/dev/parport0"
-		DEFAULT_SER_PORT="/dev/ttyS0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/parport0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/ttyS0"}
 		;;
 	*-*-linux*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="/dev/ttyS0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/ttyS0"}
 		;;
 	i[[3456]]86-*-freebsd*|amd64-*-freebsd*)
-		DEFAULT_PAR_PORT="/dev/ppi0"
-		DEFAULT_SER_PORT="/dev/cuaa0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/ppi0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/cuaa0"}
 		;;
 	*-*-freebsd*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="/dev/cuaa0"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/cuaa0"}
+		;;
+	*-*-netbsd*)
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/tty00"}
 		;;
 	*-*-solaris*)
-		DEFAULT_PAR_PORT="/dev/printers/0"
-		DEFAULT_SER_PORT="/dev/term/a"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"/dev/printers/0"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"/dev/term/a"}
 		;;
 	*-*-msdos* | *-*-mingw32* | *-*-cygwin* | *-*-windows*)
-		DEFAULT_PAR_PORT="lpt1"
-		DEFAULT_SER_PORT="com1"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"lpt1"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"com1"}
 		;;
 	*)
-		DEFAULT_PAR_PORT="unknown"
-		DEFAULT_SER_PORT="unknown"
+		DEFAULT_PAR_PORT=${DEFAULT_PAR_PORT:-"unknown"}
+		DEFAULT_SER_PORT=${DEFAULT_SER_PORT:-"unknown"}
 		;;
 esac
